### PR TITLE
(#410) "Send for approval" action in application round

### PR DESCRIPTION
**Pull request by rathorevaibhav**
#410

Changes include:

1. All actions for an application round can be selected from the drop-down (except reject). Reject will be added in the upcoming tasks.
2. Added two new actions `Send for approval` and `approve`. On send for approval, the user is asked to select a supervisor from whom the approval is requested. Approve action doesn't do anything yet.
3. A new section called "Pending" to show all applications marked as "sent-for-approval". Authorization checks haven't been implemented yet. Will follow in upcoming PRs.
4. The person requested for approval can either mark it as approved (once the approve functionality is done) or reject or schedule another round.
5. Cleanups in the `application/edit.php` to show the dropdown.

### CODE REVIEW

File: app/Models/HR/ApplicationRound.php [hunk 1]
```diff
@@ -67,6 +67,11 @@ public function _update($attr)
                 $application->reject();
                 $applicant->applications->where('id', $attr['refer_to'])->first()->markInProgress();
                 break;
+
+            case 'send-for-approval':
+                $fillable['round_status'] = 'confirmed';
```
Comment (bohraji) [inline app/Models/HR/ApplicationRound.php]: Status should be pending if the application is `send-for-approval`. @rathorevaibhav is my understanding correct here?

File: resources/views/hr/application/edit.blade.php [hunk 2]
```diff
@@ -174,43 +174,36 @@
...
                                     <button type="button" class="btn btn-primary ml-auto" data-toggle="modal" data-target="#round_{{ $applicationRound->id }}">Send mail</button>
                                 @endif
+                                </div>
                             </div>
                             @endif
                         </div>
                         <input type="hidden" name="action" value="updated">
+                        <input type="hidden" name="next_round" value="">
                         @includeWhen($applicationRound->round_status != config('constants.hr.status.confirmed.label'), 'hr.round-review-confirm-modal', ['applicationRound' => $applicationRound])
+                        @includeWhen($loop->last, 'hr.application.send-for-approval-modal')
```
Comment (bohraji) [inline resources/views/hr/application/edit.blade.php]: If this is a default model and we are not passing any value in it, why do we need to include it in the loop. Can we put it outside the loop?

File: app/Models/HR/ApplicationRound.php (hunk 1 above)
Comment (rathorevaibhav) [reply app/Models/HR/ApplicationRound.php]: @bohraji we're maintaining two statuses. One is for application and another is for an application round. For application round, the status should be confirmed as the interviewer is confirming the round and sending for supervisor's approval. The application will be marked as `sent-for-approval`.

File: resources/views/hr/application/edit.blade.php (hunk 2 above)
Comment (rathorevaibhav) [reply resources/views/hr/application/edit.blade.php]: This modal contains a submit button which should only request to update the last/latest application round. Hence, it needs to be inside the loop (and the form tags also).

Comment (rathorevaibhav) [reply resources/views/hr/application/edit.blade.php]: @bohraji let me know if that clears your understanding.

File: app/Models/HR/ApplicationRound.php (hunk 1 above)
Comment (bohraji) [reply app/Models/HR/ApplicationRound.php]: 👍

File: resources/views/hr/application/edit.blade.php (hunk 2 above)
Comment (bohraji) [reply resources/views/hr/application/edit.blade.php]: 👍

**Comment by pankaj-ag** [comment]
@rathorevaibhav I have tested all the functionalities on UAT. I also made a UI change on the dashboard. It looks good to me for production.

**Pull request merged**